packages/web3-eth-contract/src/contract.ts

```diff
@@ -602,14 +616,10 @@ export class Contract<Abi extends ContractAbi>
 			});
 		}
 
-		return new Contract<Abi>(this._jsonInterface as unknown as Abi, {
-			gas: this.options.gas,
-			gasPrice: this.options.gasPrice,
-			gasLimit: this.options.gasLimit,
-			from: this.options.from,
-			data: this.options.data,
-			provider: this.currentProvider,
-		});
+		if (this.wallet) {
+			newContract.wallet = this.wallet;
```

Comment:
This will create a new shallow copy of wallet, we should not go in that direction as:

- due to security issues ( multiple copies of wallets ) 
- if some one adding or removing accounts from main wallet it will not reflected in this sub wallet binded to contract instance?